tests/plotting/test_plot_manager.py

```diff
@@ -85,6 +89,11 @@ def trigger_remove_plot(_: Path, plot_path: str):
 
 # Note: We assign `expected_result_matched` in the callback and assert it in the test thread to avoid
 # crashing the refresh thread of the plot manager with invalid assertions.
+
+expected_result: TestRefreshResult = TestRefreshResult()
+expected_result_matched = True
```

Comment:
it seems a bit risky to keep these as global variables. I would expect the callback could capture a reference to them even if they are local to a test. Now, if a test fail to clear these the test might still pass based on the previous test result. Or the order tests run in might matter.

Reply:
Although it seems unlikely that the other tests would pass because thats not the only assert 3da3d9b should still help here to avoid it at all.